### PR TITLE
Enhanced Search

### PR DESCRIPTION
**Pull request by erceth**
Added documentation about any2any and all2any conditions on search. Added Criteria Tags. Added Single Payload Mode

### CODE REVIEW

File: docs/source/rules.rst [hunk 1]
```diff
@@ -369,6 +372,149 @@ However, the following trigger payload would not match with the ``all`` conditio
...
+      }
+    }
+
+.. warning::
+
+    When using Single Payload Mode, all2any and any2any has the same result since picking all of one thing is the 
+    same as picking any of one thing.
+
 The search operator is very powerful, but more options for the ``condition`` parameter are
 possible. At this point, only the ``any`` and ``all`` conditions are implemented, but
```
Comment (amanda11) [inline docs/source/rules.rst:519]: Thanks for the contribution (especially the corresponding st2 PR that has the new functionality). Just a minor comment on the documentation: given the addition of new conditions, then this statement below about only "any" and "all" will need updating.

Comment (erceth) [reply docs/source/rules.rst]: updated

**Comment by armab** [comment]
Looks like CI is not happy about the `.rst` changes. Could you fix?

**Comment by erceth** [comment]
Updated

**Review by armab — APPROVED** [review]
Thanks, @erceth for documentation update and @amanda11 for the review!

**Pull request merged**